text/0000-additional-exported-layers.md

```diff
@@ -0,0 +1,81 @@
+# Meta
+[meta]: #meta
+- Name: Additional exportable layers
+- Start Date: 2021-08-13
+- Author(s): [@samj1912](https://github.com/samj1912)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Allow users to provide additional paths that can be exported similar to the application directory and allow easy configuration of the application directory. These paths are defined via the `buildpack.export_dirs` by each buildpack.
+
+# Definitions
+[definitions]: #definitions
+
+- `export_dirs`: A set of directories describing where the buildpack process is likely to write data specific and should be exported alongside the application workspace. 
+
+# Motivation
+[motivation]: #motivation
+
+The main motivation behind this RFC is to unlock exporting application images that require libraries or software to be installed in certain directories apart from the default application directory and these installations are rebase safe and self-contained. This unlocks use cases where buildpacks may install software in `/opt` directories (for AWS Lambda extensions or other common standalone software) for example or preserve some settings in a `/home/$USER/.config` for the output application. This provides a rebase safe extension point without major changes to the buildpacks API.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Buildpack authors would be able to buildpacks with `buildpack.export_dirs` keys in their `buildpack.toml` files.
+
+For example a a buildpack may look like - 
+
+```toml
+api = "<buildpack API version>"
+
+[buildpack]
+id = "<buildpack ID>"
+name = "<buildpack name>"
+export_dirs = ["/opt", "/home/cnb/.config"]
```

Comment:
Are we worried about this coupling between a buildpack and the stack? Does this hurt portability of buildpacks between stacks?

Reply:
It probably will, but it is an optional feature to implement. If a buildpack is aiming for portability it could choose not to do things this way.